### ISSUE TITLE
Refactor Nexmark Source Generator as Datagen

### ISSUE THREAD

**Issue by KeXiangWang** [risingwavelabs/risingwave#2323]
The current Nexmark Source generator (https://github.com/singularity-data/risingwave/pull/540) was implemented based on the connector interface, which has serveral problems:

1. The generator is directly exposed to users, which is not a secure design for a database. So we introduce some conditional compilation setting to avoid the generator to be compiled into release version.
2. The data is generated based on a deterministic random generator, which does provide reproducible guarantee(may work differently on various platforms). So we need to consider other random generator.
3. The generator is not a connector. The better way is to introduce a new source type `datagen`, of which the nexmark generator is one of the implemenations.
4. To simulate real world usage, we should stream data into risingwave through MQ like Kafka. So a outside generator pushing data into Kafka is needed. 

To summarize, we are supposed to do two things:

- [ ] Refactor Nexmark connector as datagen.
- [ ] Introduce externel Nexmark data generator.
